hack/jenkins/e2e.sh

```diff
@@ -222,5 +222,8 @@ fi
 
 ### Clean up ###
 if [[ "${E2E_DOWN,,}" == "true" ]]; then
+    # Sleep before deleting the cluster to give the controller manager time to
+    # delete any cloudprovider resources still around from the last test.
+    sleep 60
```

Comment:
The deletions are managed using a watch and failed attempts to reconcile are retried every 5 seconds, so 60 is sort of an arbitrary bound. I've added a comment in the next push.